### PR TITLE
Add labels for sample deployment

### PR DESCRIPTION
**Pull request by shamsher31**
For better filtering of `Deployment` https://github.com/istio/istio/pull/27242#issuecomment-691468933

Moved linter fix to https://github.com/istio/istio/pull/27312

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: samples/health-check/liveness-http-same-port.yaml [hunk 1]
```diff
@@ -15,6 +15,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: liveness-http
+  labels:
+    app: liveness-http
+    version: v1
```
Comment (ericvn) [inline samples/health-check/liveness-http-same-port.yaml:20]: For consistency between all the pages, do we want the metadata name to be [app label]-[app version]?

In this case, we remove the app version or make the metadata name `liveness-http-v1`

The same comment would apply in other files.

File: samples/httpbin/httpbin-nodeport.yaml [hunk 2]
```diff
@@ -34,6 +34,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: httpbin
+  labels:
+    app: httpbin
+    version: v1
```
Comment (ericvn) [inline samples/httpbin/httpbin-nodeport.yaml:39]: As above.

File: samples/httpbin/httpbin-vault.yaml [hunk 3]
```diff
@@ -33,6 +33,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: httpbin
+  labels:
+    app: httpbin
+    version: v1
```
Comment (ericvn) [inline samples/httpbin/httpbin-vault.yaml:38]: as above

File: samples/httpbin/httpbin.yaml [hunk 4]
```diff
@@ -38,6 +38,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: httpbin
+  labels:
+    app: httpbin
+    version: v1
```
Comment (ericvn) [inline samples/httpbin/httpbin.yaml:43]: as above

File: samples/httpbin/sample-client/fortio-deploy.yaml [hunk 5]
```diff
@@ -15,6 +15,8 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: fortio-deploy
+  labels:
+    app: fortio
```
Comment (ericvn) [inline samples/httpbin/sample-client/fortio-deploy.yaml:19]: Slight mismatch on name and app.

File: samples/rawvm/k8cli.yaml.in [hunk 6]
```diff
@@ -3,6 +3,8 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: raw-cli-deployement
+  labels:
+    app: fortio
```
Comment (ericvn) [inline samples/rawvm/k8cli.yaml.in:7]: mismatch on name and app

File: samples/rawvm/k8services.yaml.in [hunk 7]
```diff
@@ -17,6 +17,8 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: raw-svc-deployment
+  labels:
+    app: echosrv
```
Comment (ericvn) [inline samples/rawvm/k8services.yaml.in:21]: mismatch on name and app

File: samples/tcp-echo/tcp-echo.yaml [hunk 8]
```diff
@@ -35,6 +35,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: tcp-echo
+  labels:
+    app: tcp-echo
+    version: v1
```
Comment (ericvn) [inline samples/tcp-echo/tcp-echo.yaml:40]: Name and app

File: samples/websockets/app.yaml [hunk 9]
```diff
@@ -15,6 +15,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: tornado
+  labels:
+    app: tornado
+    version: v1
```
Comment (ericvn) [inline samples/websockets/app.yaml:20]: name and app

File: samples/health-check/liveness-http-same-port.yaml (hunk 1 above)
Comment (shamsher31) [reply samples/health-check/liveness-http-same-port.yaml:20]: > For consistency between all the pages, do we want the metadata name to be [app label]-[app-version]?

I think that will be good to have. Not sure if that also requires doc updates wherever we use these labels?

Comment (ericvn) [reply samples/health-check/liveness-http-same-port.yaml:20]: Hence, maybe removing the version if it's not specified. I think this is making the added label(s) match the current metadata name.

**Comment by frankbu** [comment]
Actually, I'm confused why this is needed. I would have thought that the app and version labels are needed in for pod template in the Deployment, not for the Deployment itself. Note that these same labels are also needed in the `matchLabels` section of the Deployment, so now we will have the exact same labels repeated in 3 places?

cc: @douglas-reid, am I missing something?

**Comment by shamsher31** [comment]
> Note that these same labels are also needed in the matchLabels section of the Deployment, so now we will have the exact same labels repeated in 3 places?

Right, I think keeping the [same labels will be fine](https://github.com/istio/istio/blob/master/samples/bookinfo/platform/kube/bookinfo.yaml#L55). As long as `selector` labels match the `template` labels because that is how the `Deployment` will decide which `Pods` to manage.

> Actually, I'm confused why this is needed.

These labels are basically to filter out a certain type of objects from deployments.

```
$ kubectl get deployments
NAME             READY   UP-TO-DATE   AVAILABLE   AGE
details-v1       1/1     1            1           13m
productpage-v1   1/1     1            1           13m
ratings-v1       1/1     1            1           13m
reviews-v1       1/1     1            1           13m
reviews-v2       1/1     1            1           13m
reviews-v3       1/1     1            1           13m
```

Only return `version: v1`
```
$ kubectl get deployments -l version=v1
NAME             READY   UP-TO-DATE   AVAILABLE   AGE
details-v1       1/1     1            1           13m
productpage-v1   1/1     1            1           13m
ratings-v1       1/1     1            1           13m
reviews-v1       1/1     1            1           13m
```

**Comment by shamsher31** [comment]
> I would have thought that the app and version labels are needed in for pod template in the Deployment, not for the Deployment itself.

@frankbu If that is the case then the linter needs to be updated to check  Pods template `labels` and not the `Deployment` `labels` to avoid lint warnings.
```yaml
template:
    metadata:
      labels:
        app: helloworld
        version: v1
```

`Deployment` labels will be useful for filtering https://github.com/istio/istio/pull/27242#issuecomment-691468933 so I think we can keep them?

File: samples/health-check/liveness-http-same-port.yaml (hunk 1 above)
Comment (shamsher31) [reply samples/health-check/liveness-http-same-port.yaml:20]: > I think this is making the added label(s) match the current metadata name.

Please see https://github.com/istio/istio/pull/27242#issuecomment-691468933

**Comment by howardjohn** [comment]
>  If that is the case then the linter needs to be updated to check Pods template labels and not the Deployment labels to avoid lint warnings.

Yeah it seems like the linter is wrong here?

I am not against adding the labels to the deployment as well, but its a separate goal from the linter IMO

**Comment by frankbu** [comment]
I agree with @howardjohn that the linter needs to be fixed, but a separate issue.

As far as adding deployment labels is concerned, I personally think it's too much noise (i.e., repeating the same labels in 3 places in every deployment). Maybe, if we think it's particularly useful to do the kind of deployment filtering that @shamsher31 showed, above, we could add them to the specific deployments that we want to demonstrate with (e.g., productpage or helloworld). Adding it to every one of our sample deployments doesn't seem necessary, assuming the linter gets fixed.

**Comment by shamsher31** [comment]
> I agree with @howardjohn that the linter needs to be fixed, but a separate issue.

Created separate issue https://github.com/istio/istio/issues/27311 and fix for that https://github.com/istio/istio/pull/27312

**Comment by shamsher31** [comment]
> Maybe, if we think it's particularly useful to do the kind of deployment filtering that @shamsher31 showed, above, we could add them to the specific deployments that we want to demonstrate with (e.g., bookinfo or helloworld)

@frankbu Make sense. Keeping labels only for `Bookinfo`, `helloworld`, `sleep` and `tcp-echo` examples.

File: samples/websockets/app.yaml (hunk 9 above)
Comment (shamsher31) [reply samples/websockets/app.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/rawvm/k8services.yaml.in (hunk 7 above)
Comment (shamsher31) [reply samples/rawvm/k8services.yaml.in]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/rawvm/k8cli.yaml.in (hunk 6 above)
Comment (shamsher31) [reply samples/rawvm/k8cli.yaml.in]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/httpbin/sample-client/fortio-deploy.yaml (hunk 5 above)
Comment (shamsher31) [reply samples/httpbin/sample-client/fortio-deploy.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/httpbin/httpbin.yaml (hunk 4 above)
Comment (shamsher31) [reply samples/httpbin/httpbin.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/httpbin/httpbin-vault.yaml (hunk 3 above)
Comment (shamsher31) [reply samples/httpbin/httpbin-vault.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/httpbin/httpbin-nodeport.yaml (hunk 2 above)
Comment (shamsher31) [reply samples/httpbin/httpbin-nodeport.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/tcp-echo/tcp-echo.yaml (hunk 8 above)
Comment (shamsher31) [reply samples/tcp-echo/tcp-echo.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/health-check/liveness-http-same-port.yaml (hunk 1 above)
Comment (shamsher31) [reply samples/health-check/liveness-http-same-port.yaml]: Removed based on https://github.com/istio/istio/pull/27242#issuecomment-692549416

File: samples/sleep/sleep.yaml [hunk 10]
```diff
@@ -37,6 +37,8 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: sleep
+  labels:
```
Comment (frankbu) [inline samples/sleep/sleep.yaml:40]: Sleep doesn't need this, because there are no version and the label name is the same as the deployment name.

Comment (shamsher31) [reply samples/sleep/sleep.yaml]: Done.

**Review by frankbu — APPROVED** [review]

**Pull request merged**